lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -123,7 +123,8 @@ static SmallString<32> fileNameForLayer(StringRef circuitName,
 }
 
 /// For a layerblock `@A::@B::@C`, the verilog macro is `A_B_C`.
-static SmallString<32> macroNameForLayer(ArrayRef<StringAttr> layerName) {
+static SmallString<32>
+macroNameForLayer(ArrayRef<FlatSymbolRefAttr> layerName) {
```

Comment:
This change, and others that are `s/StringAttr/FlatSymbolRefAttr/` exist because the former is always trivially accessible from the latter while the latter is necessary for constructing `SymbolRefAttr`s. This became necessary due to wanting to compute macros and the `SymbolRefAttr` -> `LayerOp` mapping at the same time.